src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs

```diff
@@ -235,6 +235,105 @@ public void AuthenticationOptionsFile()
             Assert.AreEqual(1000, testCredentialsFactory.Arg2);
         }
 
+        [Test]
+        public void AuthenticationUsernamePassword()
+        {
+            const string json = @"{ ""hazelcast"": {
+""authentication"" : {
+    ""username-password"": { ""username"": ""bob"", ""password"": ""secret"" }
+}
+}}";
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(json));
```

Comment:
Could you consider to use `ReadResource(json)` for name of not to repeat parsing part? We have already a method to do that just above.

Reply:
Ah sure, have pushed a commit.